airflow/www/static/js/components/Table/Cells.tsx

```diff
@@ -150,6 +150,9 @@ export const TaskInstanceLink = ({ cell: { value, row } }: CellProps) => {
   const { sourceRunId, sourceDagId, sourceMapIndex } = row.original;
   const gridUrl = getMetaValue("grid_url");
   const dagId = getMetaValue("dag_id");
+  if (!value || !sourceRunId || !sourceDagId || !gridUrl) {
```

Comment:
Why did we change this?

Reply:
I think it's because this dataset event is created through rest API instead of any task.

**Removed**
<img width="617" alt="圖片" src="https://github.com/apache/airflow/assets/5144808/8f4b2ece-f9de-4045-afd3-c19d2d0ce764">

**Original**
<img width="604" alt="圖片" src="https://github.com/apache/airflow/assets/5144808/8633eab2-b2bc-436d-b0f9-b58d8db19221">

Reply:
perhaps we need to find a way to show that it is from the API

Reply:
![image](https://github.com/apache/airflow/assets/5144808/ac077022-e360-4eed-bdf9-0293d08444fa)

I'm thinking of adding an extra field for all such dataset event.

@dstandish @bbovenzi what do you think?

Reply:
If you rebase the extra field should already be there. But all of the checks on task instance make sense to me

Reply:
Yep, I think that column is from the latest main change. I did not change anything related to extra from the UI side

Reply:
Looks like some sort of rebase (or merge) error somewhere. I’ve reverted the change.